server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -163,7 +160,18 @@ protected final void createFieldNamesField(ParseContext context) {
 
     @Override
     public Iterator<Mapper> iterator() {
-        return multiFields.iterator();
+        Iterator<FieldMapper> multiFieldsIterator = multiFields.iterator();
```

Comment:
I'm kind of surprised that there isn't a simpler way to do this. But, :shrug:

Reply:
Java generics and inheritance continue to suck.